lightningnetwork/lnd
# scripts: use gpg --homedir flag to fix new behavior in 2.4

Fixes https://github.com/lightningnetwork/lnd/discussions/7748#discussioncomment-6145129.

## Change Description

With the latest Golang Docker base image we are using the new gpg version 2.4 is now being installed in the lnd Docker base image.

Apparently the expected value for the --keyring flag is just a file name and not an absolute path. The path of the file is indicated either by the $HOME environment variable or the --homedir flag. It looks like 2.4 now finally stopped supporting an absolute path in the --keyring flag and we need to update our gpg command to make the script work again.

This should be backward compatible and still work on older versions of gpg.

Description of change / link to associated issue.

## Steps to Test

First, checkout this PR:
```
git clone git@github.com:lightningnetwork/lnd.git
git fetch origin pull/7757/head:gpg-homedir
git checkout gpg-homedir
```

To test within Docker:

```
docker run --rm --entrypoint="" -v $(pwd)/scripts/verify-install.sh:/verify-install.sh lightninglabs/lnd:v0.16.3-beta /verify-install.sh v0.16.3-beta
```

To test standalone:

```
git checkout v0.16.3-beta
make release-install
git checkout gpg-homedir
./scripts/verify-install.sh v0.16.3-beta
```
